x-pack/plugins/reporting/server/lib/tasks/execute_report.ts

```diff
@@ -0,0 +1,420 @@
+            this.logger.info(`Stopping ${jobId}.`);
+            this.reporting.untrackReport(jobId);
+            this.logger.debug(`Reports running: ${this.reporting.countConcurrentReports()}.`);
+          } catch (failedToExecuteErr) {
+            const maxAttempts = this.config.get('capture', 'maxAttempts');
+            if (attempts > maxAttempts) {
+              // attempts remain - reschedule
+              try {
+                // reschedule to retry
+                const remainingAttempts = maxAttempts - report.attempts;
+                errorLogger(
+                  this.logger,
+                  `Scheduling retry. Retries remaining: ${remainingAttempts}.`,
+                  failedToExecuteErr
+                );
+
+                await this.rescheduleTask(reportFromTask(task).toReportTaskJSON(), this.logger);
+              } catch (rescheduleErr) {
+                // can not be rescheduled - log the error
+                errorLogger(this.logger, `Could not reschedule the errored job!`, rescheduleErr);
+              }
+            } else {
+              // 0 attempts remain - fail the job
+              try {
+                const maxAttemptsMsg = `Max attempts reached (${attempts}). Failed with: ${failedToExecuteErr}`;
+                await this._failJob(task, new Error(maxAttemptsMsg));
+              } catch (failedToFailError) {
+                errorLogger(this.logger, `Could not fail the job!`, failedToFailError);
+              }
+            }
+          }
+        },
+
+        /*
+         * Called by Task Manager to stop the report execution process in case
+         * of timeout or server shutdown
+         */
+        cancel: async () => {
+          if (jobId) {
+            this.logger.warn(`Cancelling job ${jobId}...`);
+          }
+          cancellationToken.cancel();
+        },
+      };
+    };
+  }
+
+  public getTaskDefinition() {
+    // round up from ms to the nearest second
+    const queueTimeout =
+      Math.ceil(numberToDuration(this.config.get('queue', 'timeout')).asSeconds()) + 's';
+
+    return {
+      type: REPORTING_EXECUTE_TYPE,
+      title: 'Reporting: Execute reporting jobs',
+      createTaskRunner: this.getTaskRunner(),
+      maxAttempts: 1, // NOTE: not using Task Manager retries
+      timeout: queueTimeout,
+    };
+  }
+
+  public async scheduleTask(report: ReportTaskParams) {
+    const taskInstance: ReportingExecuteTaskInstance = {
+      taskType: REPORTING_EXECUTE_TYPE,
+      state: {},
+      params: report,
+    };
+    return await this.getTaskManagerStart().schedule(taskInstance);
+  }
+
+  private async rescheduleTask(task: ReportTaskParams, logger: LevelLogger) {
+    logger.info(`Rescheduling ${task.id} to retry after error.`);
+
+    const oldTaskInstance: ReportingExecuteTaskInstance = {
+      taskType: REPORTING_EXECUTE_TYPE,
+      state: {},
+      params: task,
+    };
+    return await this.getTaskManagerStart().schedule(oldTaskInstance);
+  }
+}
```

Comment:
This file is so large that Github will not show it by default :( 

I had taken some attempts at breaking it up by making a superclass / subclass thing, but it didn't pan out as the subclass has methods that need to access private fields in the superclass, and TypeScript doesn't allow it (TS needs the `protected` keyword).